### ISSUE TITLE
how to track the loss at each epoch and use it to update learning rate

### ISSUE THREAD

**Issue by jingweimo** [fchollet/keras#4217]
I am going to use loss information to update learning rate at each epoch by the following formula:

                 learning_rate = initial_learning_rate * exp(Loss)

How to get the loss at each epoch and then implement the above updating law? Thanks!

**Comment by ishank26** [comment]
Possible duplicate #4214 

I don't know why do you want change learning rate wrt loss. IMO, that's not a good idea. Check keras [docs](https://keras.io/optimizers/#parameters-common-to-all-keras-optimizers) for adding decay. 
If you still want need this, you can use a callback-
```python
class decaylr_loss(Callback):
    def __init__(self):
        super(decaylr_loss, self).__init__()
    def on_epoch_end(self,epoch,logs={}):
        loss=logs.items()[1][1] #get loss
        print "loss: ",loss
        old_lr = self.model.optimizer.lr.get_value() #get old lr
        new_lr= old_lr*np.exp(loss) #lr*exp(loss)
        k.set_value(self.model.optimizer.lr, new_lr)
```

**Comment by jingweimo** [comment]
Thanks for your suggestion! I will test it. 

Updating learning rate wrt has the advantages of implementing a higher learning rate for higher loss , vice versa. It makes intuitive sense to improve the training performance. But it needs some effort to play around with the initial learning rate.

**Comment by jingweimo** [comment]
@ishank26 
Can I implement it using the following codes:
         
         lrate = decaylr_loss()
         model.fit(X_train, Y_train, callbacks=[lrate])


Updating learning rate wrt loss has the advantages of implementing a higher learning rate for higher loss , vice versa. It makes intuitive sense to improve the training performance. But it needs some efforts to play around with the initial learning rate.
